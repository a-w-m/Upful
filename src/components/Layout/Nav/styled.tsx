import styled from "styled-components"
import device from "../mediaQuery"
import { Link } from "gatsby"

export const Menu = styled.nav<{ open: boolean }>`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background: #f4f4f4;
  height: 100vh;
  width: 100%;
  text-align: left;
  position: fixed;
  top: 0;
  left: 0;
  transition: transform 0.3s ease-in-out;
  transform: ${({ open }) => (open ? "translateX(0)" : "translateX(-100%)")};
  //hiding slide-out-navigation to prevent focus from keyboards or screen readers
  visibility: ${({ open }) => (open ? "visible" : "hidden")};
  z-index: 9;

  @media ${device.laptop} {
    flex-direction: row;
    align-items: center;
    position: static;
    height: 2rem;
    transform: none;
    visibility: visible;
    background: none;
    gap: 2rem;
    width: 100%;
  }

  /*Remove logo from hamburger menu*/
  & > a:nth-child(1) {
    display: none;
    @media ${device.laptop} {
      display: block;
    }
  }
`

export const NavLink = styled(Link)`
  font-size: 2rem;
  text-transform: uppercase;
  padding: 2rem 0;
  padding-left: 2rem;
  font-weight: bold;
  letter-spacing: 0.5rem;
  color: #000;
  text-decoration: none;
  transition: color 0.3s linear;

  @media ${device.laptop} {
    font-size: 1.7rem;
    font-weight: 400;
  }
  &:hover {
    color: var(--header-nav-link-hover);
  }
`
