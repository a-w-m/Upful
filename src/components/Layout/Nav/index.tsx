import React from "react"
import { Menu, NavLink } from "./styled"
import Logo from "../Logo"
import { graphql, useStaticQuery } from "gatsby"
import { C } from "../../../interfaces"

interface NavProps {
  open: boolean
}

const Nav: React.FC<NavProps> = props => {
  const { open } = props

  const data: C.MenuLinks = useStaticQuery(categoryQuery)

  return (
    <Menu open={open} id = {"menu"} aria-label = "Main">
      <Logo />
      {data.site.siteMetadata.menuLinks.categories.map(category => {
        return (
          <NavLink to={category.slug} key={category.name}>
            {category.name.toUpperCase()}
          </NavLink>
        )
      })}
    </Menu>
  )
}

export const categoryQuery = graphql`
  {
    site {
      siteMetadata {
        menuLinks {
          categories {
            name
            slug
          }
        }
      }
    }
  }
`

export default Nav
