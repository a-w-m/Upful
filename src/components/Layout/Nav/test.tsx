import React from "react"
import { render, screen } from "@testing-library/react"
import Nav from "."

describe("Nav component", () => {
  it("should match snapshot", () => {
    render(<Nav open={true} />)
    const container = screen.getByRole('navigation')
    expect(container).toMatchSnapshot()
  })

  it("should not be visible when not open", () => {
    render(<Nav open={false} />)
    const navigation = screen.queryByRole('navigation', {hidden: true})
    expect(navigation).not.toBeVisible()
  })

  it("should be visible when open", () => {
    render(<Nav open={true} />)
    const navigation = screen.getByRole("navigation")
    expect(navigation).toBeVisible()
  })

  it("should contain all navigational links", ()=>{
    render(<Nav open={true} />)
    const links =  screen.getAllByRole('link')
    expect(links).toHaveLength(2)
  })

  it("should display Logo", ()=>{
    render(<Nav open={false} />)
    const logo = screen.getByAltText(/logo/)
    expect(logo).toBeInTheDocument()

  })
})
