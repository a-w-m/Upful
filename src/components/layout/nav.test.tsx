import React from "react"
import { render, screen } from "@testing-library/react"
import Nav from "./nav"


describe("Nav component", ()=>{
   
    

    test("component tree matches snapshot", ()=>{
        const open = false
        const {container} = render(<Nav open = {open}/>)
        expect(container.firstChild).toMatchSnapshot()
    })

    test("Nav is inaccessible (hidden) when open is false", ()=>{
        const open = false
        render(<Nav open = {false}/>)
        const navigation = screen.queryByRole('navigation', {hidden: true})
        expect(navigation).not.toBeVisible()

        
    })

    test("Nav is visible when open is true", ()=>{
        const open = true
        render(<Nav open = {true}/>)
        const navigation = screen.getByRole('navigation')
        expect(navigation).toBeVisible()
    })


})