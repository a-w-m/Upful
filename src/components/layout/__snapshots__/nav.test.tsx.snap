// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component component tree matches snapshot 1`] = `
<nav
  class="sc-bdfBwQ hjoKoC"
>
  <a
    href="/"
  >
    About Us
  </a>
  <a
    href="/"
  >
    Clothing
  </a>
  <a
    href="/"
  >
    Home
  </a>
  <a
    href="/"
  >
    Accessories
  </a>
</nav>
`;
